api/lib/infrastructure/datasources/airtable/challenge-datasource.js

```diff
@@ -3,6 +3,7 @@ const datasource = require('./datasource');
 const { FRENCH_FRANCE, FRENCH_SPOKEN } = require('../../../domain/constants').LOCALE;
 
 const VALIDATED_CHALLENGES = ['validé', 'validé sans test', 'pré-validé'];
+const OPERATIVE_CHALLENGES = ['validé', 'validé sans test', 'pré-validé', 'archivé'];
```

Comment:
On pourrait réutiliser la variable `VALIDATED_CHALLENGES` pour générer la variable `OPERATIVE_CHALLENGES`. Qu'en pensez-vous ?

```suggestion
const OPERATIVE_CHALLENGES = VALIDATED_CHALLENGES.concat(['archivé']);
```

Reply:
On doit même pouvoir utiliser le spread operator :
```suggestion
const OPERATIVE_CHALLENGES = [...VALIDATED_CHALLENGES, 'archivé'];
```

Reply:
Au passage, comme la nomenclature évolue, ça aurait pu être l'occasion d'ajouter une colonne (éventuellement calculée) dans Airtable qui expose vraiment l'info dont a besoin l'API à savoir `validated`/`operative` et non plus cette distinction `validé`/`validé sans test`/… qui n'a aucun intérêt en-dehors du processus de production d'épreuves… 🤔

Reply:
@jonathanperret Ne risque t'on pas d'accumuler dans Airtable des règles de métier plus que de raison ? Par exemple, on sait que la colonne PixValue dans la table d'Acquis est un genre de computed, qui contient la règle métier qu'un acquis non-actif rapporte 0 pix. Si en plus on ajoute une autre règle métier (un autre computed) pour ce que tu dis, on risque de fourrer beaucoup de logique dans Airtable (logique à ne pas oublier le jour où on veut changer de bdd ?...)

Reply:
L'exemple que tu prends — le calcule des valeurs en Pix — se trouve être codé dans Airtable au moment où nous devisons. Mais cette logique de calcul pourrait aussi être codée dans Pix Editor et que celui-ci s'occupe de maintenir les valeurs en Pix de chaque acquis à chaque changement du référentiel. C'est une problématique interne aux gens qui maintiennent le référentiel. Et ils ont aujourd'hui la liberté totale de changer la règle de calcul des Pix, ou l'implémentation de cette règle, de leur côté.

Mais pour l'API Pix qui exploite le référentiel, ça doit être transparent ! Ce qui m'embête plus que de voir des règles codées dans Airtable, c'est de voir "s'accumuler plus que de raison" 😉 dans l'API Pix des règles métier qui n'ont de sens que pour le processus interne de l'équipe de production d'épreuves.

Reply:
Je ne pense pas que Pix Editor soit la bonne place, qui est bien trop obscur selon moi. Airtable n'est pas une place de choix, mais au moins tu as de la documentation sur le sujet. À mon sens, ce serait plutôt dans une appli (base + API) dédiée à la gestion du référentiel contenu.

Reply:
On est d'accord, quand je dis "Pix Editor" je ne pense pas spécifiquement à l'outil actuel mais à l'outillage en général de la production d'épreuves. On sait qu'il devra tôt ou tard évoluer vers une application complète (back+front) de gestion du contenu.